addon/templates/components/frost-button.hbs

```diff
@@ -0,0 +1,24 @@
+{{#if isTextOnly}}
+  <div class="text">{{text}}</div>
+{{/if}}
+{{#if isIconOnly}}
+  <div class="icon">
+    {{frost-icon icon=icon}}
+  </div>
+{{/if}}
+{{#if isIconAndText}}
+  <div class="icon-text text">
+    {{frost-icon icon=icon}}
+    <div class="text">{{text}}</div>
+  </div>
+{{/if}}
+{{#if isInfo}}
+  <div class="info">
+    {{frost-icon icon=icon}}
+    <div class='content'>
+      <div class='text'>{{text}}</div>
+      <div class='subtext'>{{subtext}}</div>
+    </div>
+  </div>
+{{/if}}
+{{yield}}
```

Comment:
NOTE: I need to add integration tests to verify rendering in the different scenarios still. I'll put an issue in to make sure I don't forget.